File: 1/go/1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"sort"
	"strconv"
)

func sum(slice []int) int {
	result := 0
	for _, value := range slice {
		result += value
	}
	return result
}

func main() {
	file, err := os.Open("1/input")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()
	scanner := bufio.NewScanner(file)

	total := 0
	calories := make([]int, 0)

	for scanner.Scan() {
		if scanner.Text() == "" {
			calories = append(calories, total)
			total = 0
		} else {
			value, _ := strconv.Atoi(scanner.Text())
			total += value
		}
	}
	sort.Ints(calories)
	fmt.Println("1.1:", calories[len(calories)-1])
	fmt.Println("1.2:", sum(calories[len(calories)-3:]))
}

File: 1/python/1.py
#!/usr/bin/env python3

with open('1/input', 'r') as f:
    lines = f.readlines()

total, calories = 0, []

for line in lines:
    if line == '\n':
        calories.append(total)
        total = 0
    else:
        total += int(line.strip())

calories.sort()
print(f'1.1: {calories[-1]}')
print(f'1.2: {sum(calories[-3:])}')
